File: frontend/src/app/services/training.invitation.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {Observable} from 'rxjs';
import {BulkInvitation, TrainingInvitation, UpdateInvitation} from '../models/training.model';

@Injectable({
  providedIn: 'root',
})
export class TrainingInvitationService {
  private baseUrl = environment.apiBaseUrl + '/training/invitation';

  constructor(private http: HttpClient) {}

  getAllInvitations(): Observable<TrainingInvitation[]> {
    return this.http.get<TrainingInvitation[]>(`${this.baseUrl}`, {
    withCredentials: true,
    headers: new HttpHeaders({ 'X-Skip-Error-Handler': 'true' })
    });
  }

  inviteStaff(bulkInvitation: BulkInvitation): Observable<TrainingInvitation[]> {
    return this.http.post<TrainingInvitation[]>(`${this.baseUrl}/create`, bulkInvitation, {
      withCredentials: true,
      headers: new HttpHeaders({ 'X-Skip-Error-Handler': 'true' })
    });
  }

  getAllInvitationsByTraining(trainingId: number): Observable<TrainingInvitation[]> {
    return this.http.get<TrainingInvitation[]>(`${this.baseUrl}/get/${trainingId}`, {
      withCredentials: true,
      headers: new HttpHeaders({ 'X-Skip-Error-Handler': 'true' })
    });
  }

  getTrainingInvitationsByStaffId(staffId: string): Observable<TrainingInvitation[]> {
    return this.http.get<TrainingInvitation[]>(`${this.baseUrl}/staff/${staffId}`, {
      withCredentials: true,
      headers: new HttpHeaders({ 'X-Skip-Error-Handler': 'true' })
    });
  }

  updateInvitationStatus(invitationId: number, dto: UpdateInvitation): Observable<TrainingInvitation> {
    return this.http.patch<TrainingInvitation>(`${this.baseUrl}/${invitationId}/status`, dto, {
      withCredentials: true,
      headers: new HttpHeaders({ 'X-Skip-Error-Handler': 'true' })
    });
  }
}

File: backend/src/main/java/com/tbm/careerpathlearning/service/TrainingInvitationService.java
package com.tbm.careerpathlearning.service;

import com.tbm.careerpathlearning.dto.BulkInvitationDTO;
import com.tbm.careerpathlearning.dto.TrainingInvitationDTO;
import com.tbm.careerpathlearning.dto.UpdateInvitationDto;

import java.util.List;
import java.util.UUID;

public interface TrainingInvitationService {

    List<TrainingInvitationDTO> getTrainingInvitations();

    List<TrainingInvitationDTO> inviteStaff(BulkInvitationDTO dto, UUID userId);

    List<TrainingInvitationDTO> getAllInvitationsByTraining(Long trainingId);

    List<TrainingInvitationDTO> getTrainingInvitationsByStaffId(UUID staffId);

    TrainingInvitationDTO updateInvitationStatus(Long trainingId, UpdateInvitationDto updateInvitationDto);
}
